### PR TITLE
Improve boxing in rewrite-rules-break-SR warning + add rewrite-rules warning category

### PR DESCRIPTION
**Pull request by SkySkimmer**
also cleanup how the warning is declared

### CODE REVIEW

File: lib/cWarnings.mli [hunk 1]
```diff
@@ -122,6 +122,7 @@ module CoreCategories : sig
   val parsing : category
   val pedantic : category
   val records : category
+  val rewrite_rules : category
```
Comment (proux01) [inline lib/cWarnings.mli:125]: Should the lsit in `doc/sphinx/proof-engine/vernacular-commands.rst` be updated?

Comment (SkySkimmer) [reply lib/cWarnings.mli:125]: yes, changed

Comment (proux01) [reply lib/cWarnings.mli:125]: We may want to put a comment in that mli file to remember about that list in the doc next time.

Comment (SkySkimmer) [reply lib/cWarnings.mli:125]: There's a comment in the .ml, but the list is big enough that it doesn't appear in the diff context

**Review by proux01 — APPROVED** [review]

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
@coqbot merge now

**Pull request merged**